src/zcash/Address.hpp

```diff
@@ -0,0 +1,46 @@
+#ifndef _ZCADDRESS_H_
+#define _ZCADDRESS_H_
+
+#include "uint256.h"
+#include "serialize.h"
+
+namespace libzcash {
+
+class PaymentAddress {
+public:
+    uint256 a_pk;
+    uint256 pk_enc;
+
+    PaymentAddress() : a_pk(), pk_enc() { }
+    PaymentAddress(uint256 a_pk, uint256 pk_enc) : a_pk(a_pk), pk_enc(pk_enc) { }
+
+    ADD_SERIALIZE_METHODS;
+
+    template <typename Stream, typename Operation>
+    inline void SerializationOp(Stream& s, Operation ser_action, int nType, int nVersion) {
+        READWRITE(a_pk);
```

Comment:
I'll add the leading byte (0x92) here even though this PR isn't really addressing #572 yet.